pkg/metrics/service-monitor.go

```diff
@@ -48,6 +59,11 @@ func CreateServiceMonitors(config *rest.Config, ns string, services []*v1.Servic
 			continue
 		}
 		sm := GenerateServiceMonitor(s)
+		for k, v := range sm.ObjectMeta.Labels {
```

Comment:
See that the label with the name (`name=memcached-operator`) in the Service is applied as the Selector is already created in ServiceMonitor with it for the Operator. 

I understand that the same implementation for the customized Services and ServiceMonitor is missing and it is what you are trying to address here. 

However, IMHO the labels in the Service should be done [here](https://github.com/operator-framework/operator-sdk/blob/master/pkg/metrics/metrics.go#L80) in the same way that it is done for the operators [here](https://github.com/operator-framework/operator-sdk/blob/master/pkg/metrics/metrics.go#L115)

Reply:
Hi @camilamacedo86, thanks for reviewing this!

> See that the label with the operator name in the Service is applied as the Selector and ServiceMonitor with the Selector with for the Operator as well. So, I understand that the same implementation for the customized Services and ServiceMonitor is missing and it is what you are trying to address here.

I'm not sure I 100% follow what you're saying here, but the Service (which the ServiceMonitor monitors) isn't really relevant to my change -- I just want to be able add _additional_ labels directly to the ServiceMonitor, so that it can be matched to the correct Prometheus instance by the Prometheus operator.

> However, IMHO the labels in the Service should be done here in the same way that it is done for the operators here. Also, it would be the service name instead of labels informed then has no need for a new func.

Again, I might be understanding you incorrectly: doing things to the Service is not really necessary, but it's another way to do it (currently it seems that the ServiceMonitor gets whatever labels the Service has). To be able to create a Service with custom labels (so that the ServiceMonitor will be matched to the correct Prometheus) would require a new function also though, equivalent to [CreateMetricsService](https://github.com/operator-framework/operator-sdk/blob/5889b502a6a9a845575ba57a059cb469a4502360/pkg/metrics/metrics.go#L46) with an additional parameter for extra labels. Are you saying that I should add a new function there instead of here?

The alternative that I can see is to not have a way for people to do this, making all operators duplicate the same client setup, etc, in their main. That seems like it would be a bit of a nuisance though.

Reply:
HI @grdryn,

My IMHO is just to use the Service name as a label/selector instead of passing it as a param. In the same way that the label/selector of the operator is the name of the operator. Is it make sense?